File: c/energy_balance.c
#include <stdio.h>

int main(void) {
    double demand[] = {42, 40, 38, 36, 35, 39, 48, 55, 61, 64, 66, 67};
    double solar[] = {0, 0, 0, 0, 2, 8, 18, 34, 48, 58, 62, 64};
    double wind[] = {18, 16, 15, 14, 16, 24, 30, 26, 22, 20, 18, 16};
    int n = sizeof(demand) / sizeof(demand[0]);
    double total_demand = 0.0;
    double total_renewable = 0.0;

    for (int i = 0; i < n; i++) {
        total_demand += demand[i];
        total_renewable += solar[i] + wind[i];
    }

    printf("C energy balance summary\n");
    printf("Total demand: %.2f MWh\n", total_demand);
    printf("Total renewable generation: %.2f MWh\n", total_renewable);
    printf("Renewable share proxy: %.3f\n", total_renewable / total_demand);
    return 0;
}

File: rust/energy_balance.rs
fn main() {
    let demand = [42.0, 40.0, 38.0, 36.0, 35.0, 39.0, 48.0, 55.0, 61.0, 64.0, 66.0, 67.0];
    let solar =  [0.0, 0.0, 0.0, 0.0, 2.0, 8.0, 18.0, 34.0, 48.0, 58.0, 62.0, 64.0];
    let wind =   [18.0, 16.0, 15.0, 14.0, 16.0, 24.0, 30.0, 26.0, 22.0, 20.0, 18.0, 16.0];

    let total_demand: f64 = demand.iter().sum();
    let total_renewable: f64 = solar.iter().zip(wind.iter()).map(|(s, w)| s + w).sum();

    println!("Rust energy balance summary");
    println!("Total demand: {:.2} MWh", total_demand);
    println!("Total renewable generation: {:.2} MWh", total_renewable);
    println!("Renewable share proxy: {:.3}", total_renewable / total_demand);
}
